presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/TestHiveClient.java

```diff
@@ -16,15 +16,17 @@
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Parameters;
 import org.testng.annotations.Test;
+import org.xbill.DNS.Lookup;
 
 @Test
 public class TestHiveClient
         extends AbstractTestHiveClient
 {
-    @Parameters({"hive.hadoop2.metastoreHost", "hive.hadoop2.metastorePort", "hive.hadoop2.databaseName", "hive.hadoop2.timeZone"})
+    @Parameters({"hive.hadoop2.metastoreHost", "hive.hadoop2.metastorePort", "hive.hadoop2.databaseName", "hive.hadoop2.timeZone", "dns.port"})
     @BeforeClass
-    public void initialize(String host, int port, String databaseName, String timeZone)
+    public void initialize(String host, int port, String databaseName, String timeZone, int dnsPort)
     {
+        Lookup.getDefaultResolver().setPort(dnsPort);
```

Comment:
As discussed with @cberner, the goal is to execute integration tests on native machine while Hive is dockerized. In order for tests to communicate with Hive and Hdfs, a SOCKS proxy is setup, which is used by Hive connector. Additionally, we need a DNS server, which resolves Docker cluster IP addresses (e.g. hadoop-master). This DNS is accessible from native machine (though Docker port forwarding). It cannot be forwarded to standard DNS port (53) though, because on Travis this port is already bound. We forward DNS server to port 5353.

When launching integration tests, we set JVM to use our DNS server (by using dnsjava provider). We have to change port though, because by default port 53 is used.